### PR TITLE
Updated to use React.createRef

### PR DESCRIPTION
**Pull request by mdabydeen**
#### Summary
Migrates the string ref for the security tab in the admin console. 

#### Ticket Link
fixes https://github.com/mattermost/mattermost-server/issues/15805

### CODE REVIEW

**Comment by jasonblais** [comment]
Hey @mdabydeen, thanks for your contribution! It looks like the tests are failing, would you be open to take a look? We'd be happy to help!

File: components/user_settings/security/user_access_token_section/user_access_token_section.jsx [hunk 1]
```diff
@@ -78,37 +78,38 @@ export default class UserAccessTokenSection extends React.PureComponent {
...
     }
 
     stopCreatingToken = () => {
-        this.setState({tokenCreationState: TOKEN_NOT_CREATING, saving: false});
+        this.setState({ tokenCreationState: TOKEN_NOT_CREATING, saving: false });
     }
 
     handleCreateToken = async () => {
         this.handleCancelConfirm();
+        this.newtokendescription = React.createRef();
```
Comment (Willyfrog) [inline components/user_settings/security/user_access_token_section/user_access_token_section.jsx:94]: Can you rename for consistency with other PRs?
```suggestion
        this.newtokendescriptionRef = React.createRef();
```

**Review by Willyfrog — CHANGES_REQUESTED** [review]
hi there, thanks for contributing! 
Seems like something auto-formated parts of the file, can you revert those changes so the linting passes?
can you also rename the reference for consistency with other PRs?

File: components/user_settings/security/user_access_token_section/user_access_token_section.jsx [hunk 2]
```diff
@@ -4,12 +4,12 @@
 
 import PropTypes from 'prop-types';
 import React from 'react';
-import {FormattedMessage} from 'react-intl';
+import { FormattedMessage } from 'react-intl';
```
Comment (nevyangelova) [inline components/user_settings/security/user_access_token_section/user_access_token_section.jsx:7]: Nit: would it be ok to set your formatter to match the Mattermost style? A lot of these changes are unrelated to the PR.

**Review by nevyangelova — CHANGES_REQUESTED** [review]

File: components/user_settings/security/user_access_token_section/user_access_token_section.jsx (hunk 2 above)
Comment (mdabydeen) [reply components/user_settings/security/user_access_token_section/user_access_token_section.jsx]: @nevyangelova thanks. I didn't realize my editor messed up. It's fixed now

**Comment by mdabydeen** [comment]
@jasonblais all tests are passing now.

**Review by Willyfrog — CHANGES_REQUESTED** [review]
thanks for taking care of the linting, this looks much better and it is easier to review.

at https://github.com/mattermost/mattermost-webapp/pull/6648/files#diff-2d48de396357e98d63b088f183c2bb1eR438 you forgot to actually set the reference so it can be used.

**Comment by mdabydeen** [comment]
> at https://github.com/mattermost/mattermost-webapp/pull/6648/files#diff-2d48de396357e98d63b088f183c2bb1eR438 you forgot to actually set the reference so it can be used.

@Willyfrog i'm not sure what is meant by this. I thought it was already taken care of. Can you provider some clarity?

**Comment by Willyfrog** [comment]
> > at https://github.com/mattermost/mattermost-webapp/pull/6648/files#diff-2d48de396357e98d63b088f183c2bb1eR438 you forgot to actually set the reference so it can be used.
> 
> @Willyfrog i'm not sure what is meant by this. I thought it was already taken care of. Can you provider some clarity?

sorry, I gave you the wrong link, I meant here:
https://github.com/mattermost/mattermost-webapp/blob/1d44308d0d3f0e51f7311a93b3537a5e8e01718a/components/user_settings/security/user_access_token_section/user_access_token_section.jsx#L438
it is still using the string references, when it should be using `this.newtokendescriptionRef`

Also, could you remove the comment for disabling the eslint rule? that way we can prevent people from adding string references back to this file. It looks like:
`/* eslint-disable react/no-string-refs */`

**Comment by mdabydeen** [comment]
@Willyfrog all changes updated

**Review by nevyangelova — APPROVED** [review]

**Review by Willyfrog — APPROVED** [review]
LGTM! thanks for contributing!

**Review by stevemudie — APPROVED** [review]
LGTM
